var/spack/repos/builtin/packages/neuron/package.py

```diff
@@ -0,0 +1,236 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+import os
+import sys
+from spack import *
+
+
+class Neuron(Package):
+
+    """NEURON is a simulation environment for modeling individual
+    and networks of neurons. NEURON models individual neurons via
+    the use of sections that are automatically subdivided into individual
+    compartments, instead of requiring the user to manually create
+    compartments. The primary scripting language is hoc but a Python
+    interface is also available."""
+
+    homepage = "https://www.neuron.yale.edu/"
+    url      = "http://www.neuron.yale.edu/ftp/neuron/versions/v7.4/nrn-7.4.tar.gz"
+    github   = "https://github.com/nrnhines/nrn"
+
+    version('7.4', '2c0bbee8a9e55d60fa26336f4ab7acbf')
+    version('7.3', '993e539cb8bf102ca52e9fefd644ab61')
+    version('7.2', '5486709b6366add932e3a6d141c4f7ad')
+    version('develop', git=github, preferred=True)
+
+    variant('mpi',           default=True,  description='Enable MPI parallelism')
+    variant('python',        default=True,  description='Enable python')
+    variant('static',        default=True,  description='Build static libraries')
```

Comment:
Most packages have a `shared` variant. `+shared` implies shared libraries, `~shared` implies static libraries. For comparison:
```
$ grep variant.*shared */package.py | wc -l
      53
$ grep variant.*static */package.py | wc -l
       2
```